tests/testthat/test-create_trends_ensemble.R

```diff
@@ -0,0 +1,215 @@
+)
+
+daily_ts <- expand.grid(
+  stringsAsFactors = FALSE,
+  location = c("ak", "al"),
+  time_index = as.Date("2022-11-05") + 1:28,
+  observation = NA
+)
+daily_ts$observation[daily_ts$location == "ak"] <-
+  c(8, 9, 6, 7, 3, 6, 6, 5, 4, 11, 10, 3, 4, 3,
+    3, 8, 6, 7, 10, 6, 4, 4, 6, 5, 2, 5, 6, 5)
+daily_ts$observation[daily_ts$location == "al"] <-
+  c(27, 19, 20, 16, 19, 22, 20, 21, 18, 25, 17, 25, 27, 22,
+    15, 32, 21, 26, 18, 14, 14, 14, 24, 23, 16, 20, 14, 44)
+
+
+test_that("missing or extraneous columns in component_variations throws an error", {
+  daily_variations[, 1] |>
+    create_trends_ensemble(daily_ts,
+                           reference_date = "2022-12-10",
+                           horizons = -6:21,
+                           target = "inc hosp",
+                           quantile_levels = c(.1, .5, .9),
+                           n_samples = NULL,
+                           return_baseline_predictions = FALSE) |>
+    expect_error(regex = "`component_variations` is missing the column",
+                 fixed = TRUE)
+
+  daily_variations |>
+    dplyr::mutate(horizons = 28) |>
+    create_trends_ensemble(daily_ts,
+                           reference_date = "2022-12-10",
+                           horizons = -6:21,
+                           target = "inc hosp",
+                           quantile_levels = c(.1, .5, .9),
+                           n_samples = NULL,
+                           return_baseline_predictions = FALSE) |>
+    expect_error(regex = "`component_variations` contains the extra column",
+                 fixed = TRUE)
+})
+
+test_that("unsupported temporal_resolution values in component_variations throws an error", {
+  daily_variations |>
+    dplyr::mutate(temporal_resolution = "monthly") |>
+    dplyr::bind_rows(daily_variations) |>
+    create_trends_ensemble(daily_ts,
+                           reference_date = "2022-12-10",
+                           horizons = -6:21,
+                           target = "inc hosp",
+                           quantile_levels = c(.1, .5, .9),
+                           n_samples = NULL,
+                           return_baseline_predictions = FALSE) |>
+    expect_error(regex = "`component_variations` must only include temporal resolution values",
+                 fixed = TRUE)
+})
+
+test_that("multiple temporal_resolution values in component_variations throws an error", {
+  daily_variations |>
+    dplyr::bind_rows(weekly_variations) |>
+    create_trends_ensemble(daily_ts,
+                           reference_date = "2022-12-10",
+                           horizons = -6:21,
+                           target = "inc hosp",
+                           quantile_levels = c(.1, .5, .9),
+                           n_samples = NULL,
+                           return_baseline_predictions = FALSE) |>
+    expect_error(regex = "Currently `component_variations` may only contain one unique temporal resolution value",
```

Comment:
Low priority -- is this true? I thought I saw stuff about splitting by the temporal resolution up above.

Reply:
Yes, because the current ensembling for samples makes it so that having multiple temporal resolution values results in different numbers of samples per model, but I wanted to put the check closer to the top to avoid unnecessary calculations. This validation will be removed later on once support is added, but I figured I would just put it in until then